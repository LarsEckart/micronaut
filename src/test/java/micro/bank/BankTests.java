package micro.bank;

import javax.inject.Inject;

import io.micronaut.runtime.server.EmbeddedServer;
import io.micronaut.test.annotation.MicronautTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;
import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;

@MicronautTest
class BankTests {

  @Inject EmbeddedServer server;

  @Test
  void all_accounts_after_migrations() {
    verifyPath("/account");
  }

  @Test
  void single_account_info() {
    verifyPath("/account/EE1420041010050500013M02606");
  }

  private void verifyPath(String path) {
    String response = given()
        .port(server.getPort())
        .when()
        .get(path)
      .then()
        .extract()
        .asString();

    Approvals.verifyJson(response);
  }
}
