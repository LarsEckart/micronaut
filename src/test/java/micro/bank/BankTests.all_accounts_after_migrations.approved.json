[
  {
    "amount": 111.11,
    "iban": "EE1420041010050500013M02606"
  },
  {
    "amount": 222.22,
    "iban": "FI89370400440532013000"
  }
]